web/src/engine/predictive-text/worker-thread/src/main/correction/tokenization-subsets.ts

```diff
@@ -0,0 +1,83 @@
+import { SENTINEL_CODE_UNIT } from '@keymanapp/models-templates';
+import { KMWString } from '@keymanapp/web-utils';
+
+import { PrecomputedTokenizationTransition } from './context-tokenization.js';
+
+export function precomputationSubsetKeyer(precomputation: PrecomputedTokenizationTransition): string {
+  const { tokenMapping, transformMap } = precomputation;
+  const { edgeWindow, merges, splits, unmappedEdits } = tokenMapping;
+  const components: string[] = [];
+
+  // First entry: based on the edge window.  The real key:  what's the edit
+  // boundary?  We need to apply to the same token and portion thereof.
+  const editBoundary = edgeWindow.editBoundary;
+
+  // It's not about the boundary text - we just need to ensure it's the 'same'
+  // token - comprised of the same keystrokes.  `sourceText` reflects the actual
+  // input for the source keystrokes.  We might have deleted part of it in this
+  // tokenization, but that doesn't matter here - we want to imply the
+  // represented keystroke range.
+  const boundaryEdgeIndex = editBoundary.tokenIndex - edgeWindow.sliceIndex;
+  const boundaryComponent = `B${editBoundary.tokenIndex}=${editBoundary.sourceText}`;
```

Comment:
Self-note:  rework the references to `.sourceText`.  (The comment above also appears, copied, in the "legacy keyer" followup.)